### ISSUE TITLE
One out-of-vocab token id turns the whole window's logits non-finite, including positions before it

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#233]
Found while building the non-finite guard for #229. **Not the #229 failure** — that one
happens on in-vocab input — but the same silent shape, and worth its own record.

## Measured

Toy `RefinerForTraining` (dim 32, vocab 37, 1 encoder layer), a full 512-token window of
valid ids, then the same window with **one** id set to 50256:

```
all ids in vocab                       non-finite      0 / 18944
ONE id out of vocab (50256) at pos 7   non-finite  18944 / 18944
```

One bad id poisons **the entire output**, including positions 0–6, which are causally
*before* it. So this does not travel through attention — something global (a norm, or the
shared refine state) spreads it across the whole window.

The embedding gather itself is not the culprit: JAX clamps an out-of-range index rather
than filling with NaN (verified — index 50256 into a 37-row table returns row 36's
values, finite).

## Why it is currently harmless, and why that is not a reason to leave it

`r50k_base` emits ids ≤ 50256 and `VOCAB_SIZE` is 50304, so production ids are always in
range. The exposure is:

- **A future tokenizer swap** where the padding to a round vocab is forgotten.
- **A corrupted data shard.** The base run skipped 14,163 non-finite steps (0.36%),
  attributed to f16 overflow. This is a *second* mechanism that produces the same
  telemetry, and nothing distinguishes them today. Stated as a hypothesis worth ruling
  out, not a claim — no shard has been checked.
- **Any caller passing ids from outside the pipeline**, which is what found this.

## What to do

1. Fail loudly on an out-of-range id at the model boundary rather than returning NaN —
   an assert in the toy/test path at minimum, since the production path can afford no
   per-step check.
2. Find *where* it becomes global. Positions before the bad token going NaN is the
   informative part, and pins whether it is a norm or the refine state. This is a good
   first customer for `instruments/latents.py` (#225) once that exists.
3. Cheap corroboration for the skip-count hypothesis: scan the tokenized corpus for any
   id ≥ `VOCAB_SIZE`. One pass over `runs/data/`, no GPU. If it comes back clean, the
   hypothesis dies and this stays a robustness nit.

Related: #229 (the guard, which now surfaces this as a clear error rather than fluent
nonsense), #225.
